### PR TITLE
COMMERCE-5514 As an admin, I want to select an address validation service in System Settings

### PR DESCRIPTION
**Pull request by katienesterovich**
### [COMMERCE-5514](https://issues.liferay.com/browse/COMMERCE-5514) As an admin, I want to select an address validation service in System Settings

Depends on https://github.com/liferay-commerce/liferay-portal/pull/560

1. Created a configuration category for Address Validation under Commerce 
image

2. Created a `General` configuration for Address Validation
image

3. Created an Address Validation Engine Registry 
4. Created an Address Validation Engine Options Provider which uses the above (3) engineRegistry to populate the `Address Validation Engine` configuration field in (2). 

Needs a buildLang commit.

### CODE REVIEW

**Comment by katienesterovich** [comment]
ci:test:sf

**Comment by katienesterovich** [comment]
ci:test:relevant

File: modules/dxp/apps/commerce/commerce-address-validation-engine-api/src/main/java/com/liferay/commerce/address/validation/engine/configuration/CommerceAddressValidationEngineConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		deflt = StringPool.FALSE, name = "validate-address-in-warehouses-page",
+		required = false
+	)
+	public boolean validateAddressInWarehousesPage();
+
+	@Meta.AD(
+		deflt = StringPool.FALSE, name = "validate-address-in-shipments-widget",
+		required = false
+	)
+	public boolean validateAddressInWidgetsPage();
```
Comment (dsitu) [inline modules/dxp/apps/commerce/commerce-address-validation-engine-api/src/main/java/com/liferay/commerce/address/validation/engine/configuration/CommerceAddressValidationEngineConfiguration.java:71]: I think it should be named this.
```suggestion
	public boolean validateAddressInShipmentsWidgetsPage();
```

File: modules/dxp/apps/commerce/commerce-address-validation-engine-api/src/main/java/com/liferay/commerce/address/validation/engine/configuration/CommerceAddressValidationEngineConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		deflt = StringPool.FALSE, name = "validate-address-in-orders-page",
+		required = false
+	)
+	public boolean validateAddressInOrdersPage();
+
+	@Meta.AD(
+		deflt = StringPool.FALSE, name = "validate-address-in-invoicing-page",
+		required = false
+	)
+	public boolean validateAddressInInvoivingPage();
```
Comment (dsitu) [inline modules/dxp/apps/commerce/commerce-address-validation-engine-api/src/main/java/com/liferay/commerce/address/validation/engine/configuration/CommerceAddressValidationEngineConfiguration.java:83]: typo
```suggestion
	public boolean validateAddressInInvoicingPage();
```

File: modules/dxp/apps/commerce/commerce-address-validation-engine/src/main/java/com/liferay/commerce/address/validation/engine/internal/configuration/definition/CommerceAddressValidationEngineConfigurationFieldOptionsProvider.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.commerce.address.validation.engine.internal.configuration.definition;
```
Comment (dsitu) [inline modules/dxp/apps/commerce/commerce-address-validation-engine/src/main/java/com/liferay/commerce/address/validation/engine/internal/configuration/definition/CommerceAddressValidationEngineConfigurationFieldOptionsProvider.java:15]: I think this should match the package path of ConfigurationFieldOptionsProvider.

```suggestion
package com.liferay.commerce.address.validation.engine.internal.configuration.admin.definition;
```

File: modules/dxp/apps/commerce/commerce-address-validation-engine/src/main/java/com/liferay/commerce/address/validation/engine/internal/configuration/definition/CommerceAddressValidationEngineConfigurationFieldOptionsProvider.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+					return commerceAddressValidationEngine.
+						getAddressValidationEngineName();
+				}
+
+			};
+
+			options.add(option);
+		}
+
+		return options;
```
Comment (dsitu) [inline modules/dxp/apps/commerce/commerce-address-validation-engine/src/main/java/com/liferay/commerce/address/validation/engine/internal/configuration/definition/CommerceAddressValidationEngineConfigurationFieldOptionsProvider.java:74]: Sort, wrap like a burrito.

```suggestion
		List<Option> options = new ArrayList<>();

		Map<String, CommerceAddressValidationEngine>
			commerceAddressValidationEngines =
				_commerceAddressValidationEngineRegistry.
					getCommerceAddressValidationEngines();

		for (CommerceAddressValidationEngine commerceAddressValidationEngine :
				commerceAddressValidationEngines.values()) {

			Option option = new Option() {

				@Override
				public String getLabel(Locale locale) {
					return commerceAddressValidationEngine.
						getAddressValidationEngineName();
				}

				@Override
				public String getValue() {
					return commerceAddressValidationEngine.
						getAddressValidationEngineName();
				}

			};

			options.add(option);
		}

		return options;
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by katienesterovich** [comment]
ci:test:sf

**Comment by katienesterovich** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

**Comment by katienesterovich** [comment]
ci:test:relevant

**Comment by katienesterovich** [comment]
Rebased in https://github.com/liferay-commerce/liferay-portal/pull/888

**Pull request closed without merging**